racksdb/web/ui/src/views/InfrastructureDetailsView.vue

```diff
@@ -0,0 +1,161 @@
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http'
+import { ref, onMounted, inject } from 'vue'
+import SearchBar from '@/components/SearchBar.vue'
+import InfrastructureCards from '@/components/InfrastructureCards.vue'
+import { injectionKey } from '@/plugins/runtimeConfiguration';
+import type { Ref } from 'vue'
+import type { Infrastructure } from '@/views/InfrastructuresView.vue'
+
+const http = useHttp()
+const infrastructures: Ref<Array<Infrastructure>> = ref([])
+const infrastructureDetails: Ref<Infrastructure | undefined> = ref()
+const showFullImg = ref(false)
+const rack = ref()
+const cardsView = ref(true)
+const tableView = ref(false)
+const showRack = ref(false)
+const showInfrastructureRacks = ref(true)
+
+function choseView(){
+    if(cardsView.value){
+        cardsView.value = !cardsView.value
+        tableView.value = !tableView.value 
+    }else{
+        tableView.value = !tableView.value
+        cardsView.value = !cardsView.value
+    }
+}
+
+function openImg(){
```

Comment:
As well as for `DatacenterRoomView`, this function is also used to close image modal. A better name would be `toggleImageModal()`.